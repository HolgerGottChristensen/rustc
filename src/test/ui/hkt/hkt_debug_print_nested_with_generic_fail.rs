// HKT test that something cloneable can be called with clone, method call syntax, definition and call.
// check-fail

use std::fmt::Debug;

fn test<T, I<%J: Debug>: Debug>(input: I<T>) { //~ ERROR: `T` doesn't implement `Debug`
    println!("Hej verden: {:?}", input);
}

fn main() {
    test::<u32, Option<%J>>(Some(42));
}
