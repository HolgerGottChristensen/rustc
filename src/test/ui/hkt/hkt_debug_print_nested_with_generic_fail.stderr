error[E0277]: `T` doesn't implement `Debug`
  --> $DIR/hkt_debug_print_nested_with_generic_fail.rs:6:40
   |
LL | fn test<T, I<%J: Debug>: Debug>(input: I<T>) {
   |                                        ^^^^ `T` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
note: required by a bound in `test::I`
  --> $DIR/hkt_debug_print_nested_with_generic_fail.rs:6:18
   |
LL | fn test<T, I<%J: Debug>: Debug>(input: I<T>) {
   |                  ^^^^^ required by this bound in `test::I`
help: consider restricting type parameter `T`
   |
LL | fn test<T: std::fmt::Debug, I<%J: Debug>: Debug>(input: I<T>) {
   |          +++++++++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
